react/components/Conditions/index.tsx

```diff
@@ -0,0 +1,265 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import Button from '../Button/index'
+import ButtonWithIcon from '../ButtonWithIcon/index'
+import IconPlus from '../icon/Plus/index'
+import IconClose from '../icon/Close/index'
+import Separator from './Separator'
+import StrategySelector from './StrategySelector'
+import Statement from '../Statement/index'
+
+const MEDIUM_ICON_SIZE = 16
+
+const propTypes = {
+  /** Shows or hides the delete button */
+  canDelete: PropTypes.bool,
+  /** Wether to show this component stretched to the width */
+  isFullWidth: PropTypes.bool,
+  /** Whether the order of elements and text if right to left */
+  isRtl: PropTypes.bool,
+  /** Labels for the controls and texts, default is english */
+  labels: PropTypes.shape({
+    addNewCondition: PropTypes.string,
+    addConditionBtn: PropTypes.string,
+    delete: PropTypes.string,
+    noConditions: PropTypes.string,
+    operatorAll: PropTypes.string,
+    operatorAnd: PropTypes.string,
+    operatorAny: PropTypes.string,
+    operatorOr: PropTypes.string,
+    headerPrefix: PropTypes.string,
+    headerSufix: PropTypes.string,
+  }),
+  /** Conditions change callback: array of statement definitions */
+  onChangeStatements: PropTypes.func.isRequired,
+  /** Operator (any, all) change callback  */
+  onChangeOperator: PropTypes.func,
+  /** Operator indicates whether all the statements should be met or any of them */
+  operator: PropTypes.oneOf(['all', 'any']),
+  /** Possible options and respective data types, verb options */
+  options: PropTypes.object.isRequired,
+  /** Show or hide the header that selects the operator (any vs all) */
+  showOperator: PropTypes.bool,
+  /** Current selected options for all statements */
+  statements: PropTypes.arrayOf(
+    PropTypes.shape({
+      subject: PropTypes.string,
+      verb: PropTypes.string,
+      object: PropTypes.any,
+      error: PropTypes.any,
+    })
+  ),
+  /** Placeholder for subject dropdown */
+  subjectPlaceholder: PropTypes.string.isRequired,
+}
+
+type Props = PropTypes.InferProps<typeof propTypes>
+
+const Conditions: React.FC<Props> = ({
+  canDelete,
+  statements,
+  options,
+  subjectPlaceholder,
+  isFullWidth,
+  isRtl,
+  labels,
+  showOperator,
+  operator,
+  onChangeOperator,
+  onChangeStatements,
+}) => {
+  const objectIsEmpty = object => {
```

Comment:
All these conditions are considered `falsy`, so this means that instead of using `objectIsEmpty(yourObject)` you could do `!!yourObject`. right? 🤔

Reply:
oh, just the empty array that isn't :/

Reply:
Ow, yeah =/. Forgot that